core/capability-negotiation.md

```diff
@@ -0,0 +1,457 @@
+
+### The CAP DEL subcommand
+
+The NEW subcommand MUST ONLY be sent to clients that have negotiated CAP Version `302` or
+enabled the `cap-notify` capability.
+
+The DEL subcommand signals that the server no longer supports one or more capabilities that
+have been advertised. Upon receiving a `CAP DEL` message, the client MUST treat the listed
+capabilities as cancelled and no longer available. Clients SHOULD NOT send `CAP REQ`
+messages to cancel the capabilities in `CAP DEL`, as they have already been cancelled by
+the server.
+
+Servers MUST cancel any capability-specific behavior for a client after sending the
+`CAP DEL` message to the client.
+
+Clients MUST gracefully handle situations when the server removes support for any
+capability. If the client cannot continue to operate without a capability, disconnecting
+with an appropriate `QUIT` message is acceptable.
+
+The format of a `CAP DEL` message is:
+
+    CAP <nick> DEL :<extension 1> [<extension 2> ... [<extension n>]]
+
+The last parameter is a space-separated list of capabilities that are no longer available.
+
+Example:
+
+    Server: :irc.example.com CAP modernclient DEL :userhost-in-names multi-prefix away-notify
+
+## `cap-notify`
+
+The `cap-notify` capability indicates support for the `NEW` and `DEL` messages listed above.
+This capabilitiy MUST be implicitly enabled if the client requests `CAP LS` with a version
+of 302 or newer (`CAP LS 302`), as described above.
+
+Further, the `cap-notify` capability MAY NOT be disabled if the client requests `CAP LS`
+with a version of 302 or newer.
+
+When implicitly enabled via this mechanism, servers MAY list the `cap-notify` capability
+in `CAP LS` and `CAP LIST` responses. Additionally client MAY request the capability with
+`CAP REQ`, and capable servers MUST accept and `CAP ACK` the request without side effects.
+This lets clients blindly enable this capability, regardless of it being implicitly
+enabled by the server.
+
+Clients that do not support CAP Version 302 MAY request the `cap-notify` capability
+explicitly. Such clients MAY disable the capability at any time.  This to ease the
+adaptation of new features.
+
+When enabled, the server MUST notify the client about all new capabilities and about
+existing capabilities that are no longer available using the `NEW` and `DEL` subcommands
+listed above.
+
+## Rules for naming capabilities
+
+The full capability name MUST be treated as an opaque identifier.
+
+There are two capability namespaces:
+
+### Vendor-Specific
+
+Names which contain a slash character (`/`) designate a vendor-specific capability namespace.
+
+These names are prefixed by a valid DNS domain name.
+
+For example: `znc.in/server-time`.
+
+In cases where the prefix contains non-ASCII characters, punycode MUST be used,
+e.g. `xn--e1afmkfd.org/foo`.
+
+Vendor-Specific capabilities should be submitted to the IRCv3 working group for consideration.
+
```

Comment:
We need to include the `draft/` namespace as reserved for standardisation and listed in the registry.

Reply:
Roughly wrote this out, hopefully this works.